test/cpp_extensions/python_agnostic_extension/setup.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+import distutils.command.clean
+import shutil
+from pathlib import Path
+
+from setuptools import setup
+
+from torch.utils.cpp_extension import BuildExtension, CUDAExtension
+
+
+ROOT_DIR = Path(__file__).parent
+CSRC_DIR = ROOT_DIR / "python_agnostic" / "csrc"
+
+
+class clean(distutils.command.clean.clean):
+    def run(self):
+        # Run default behavior first
+        distutils.command.clean.clean.run(self)
+
+        # Remove extension
+        for path in (ROOT_DIR / "python_agnostic").glob("**/*.so"):
+            path.unlink()
+        # Remove build and dist and egg-info directories
+        dirs = [
+            ROOT_DIR / "build",
+            ROOT_DIR / "dist",
+            ROOT_DIR / "python_agnostic.egg-info",
+        ]
+        for path in dirs:
+            if path.exists():
+                shutil.rmtree(str(path), ignore_errors=True)
+
+
+def get_extension():
+    extra_compile_args = {
+        "cxx": [
+            "-O3",
+            "-fdiagnostics-color=always",
+        ],
+        "nvcc": [
+            "-O3",
+            "-t=0",
```

Comment:
Shouldn't this be the default?

Reply:
I actually cannot find a manpage explaining -t=0 (do you know what that is?) for nvcc. In either case, having compiler flags that I don't understand for a unit test is silly, so I'm removing the flags.

Reply:
https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/#threads-number-t